### PR TITLE
feat: add survey schedule option

### PR DESCRIPTION
**Pull request by mattinannt**
## What does this PR do?

This PR adds the option to schedule a survey for a later so it gets published on the provided date automatically.
Screenshot 2024-04-03 at 18 05 59

Fixes # (issue)

## How should this be tested?

- Create a new survey and set the new scheduling option in the response settings
- Once published, the survey should have the status `scheduled`.
- Change the `runOnDate` field in the survey manually to date in the past using prisma studio
- Run the cron job by calling the `/api/cron/survey-status` endpoint with a valid cron secret
```
curl --location --request POST 'http://localhost:3000/api/cron/survey-status' \
--header 'X-Api-Key: MY_SECRET'
```
- The survey should now be published

### CODE REVIEW

**Review by jobenjada — APPROVED** [review]
UI looks great! Didn't test the functionality so @gupta-piyush19 pls do so before merging :)

File: apps/web/app/api/cron/survey-status/route.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+export async function POST() {
+  const headersList = headers();
+  const apiKey = headersList.get("x-api-key");
+
+  if (!apiKey || apiKey !== CRON_SECRET) {
+    return responses.notAuthenticatedResponse();
+  }
+
+  // close surveys that are in progress and have a closeOnDate in the past
+  const surveys = await prisma.survey.findMany({
```
Comment (gupta-piyush19) [inline apps/web/app/api/cron/survey-status/route.ts:16]: can we rename the `surveys` variable to a more descriptive name, like `surveysToClose`

File: .github/workflows/cron-closeOnDate.yml
Comment (gupta-piyush19) [inline .github/workflows/cron-closeOnDate.yml:1]: can we rename this file to `cron-surveyStatusUpdate`? Also, please check in other CRON(s) to follow the same convention(name with spaces in the first line).

File: docker/cronjobs [hunk 2]
```diff
@@ -1,3 +1,3 @@
-0 0 * * * curl $WEBAPP_URL/api/cron/close_surveys -X POST -H 'content-type: application/json' -H 'x-api-key: '"$CRON_SECRET"''
+0 0 * * * curl $WEBAPP_URL/api/cron/survey-status -X POST -H 'content-type: application/json' -H 'x-api-key: '"$CRON_SECRET"''
 0 8 * * 1 curl $WEBAPP_URL/api/cron/weekly_summary -X POST -H 'content-type: application/json' -H 'x-api-key: '"$CRON_SECRET"''
```
Comment (gupta-piyush19) [inline docker/cronjobs:2]: Can we make it consistent to use either hyphen(-) or underscore(_)?
weekly_summary
close_survey -> survey-status

**Review by gupta-piyush19 — CHANGES_REQUESTED** [review]
Thanks, @mattinannt, for the feature; it works really well 🚀 
I just have a few suggestions related to code consistency; it would be great if you could comment on or fix them. 
Also, one more question: If a survey is already completed and I try to set the runOnDate, should it move to scheduled status or not? Currently, it's not.

**Comment by mattinannt** [comment]
@gupta-piyush19 Thanks for the review; I made the required changes; please take a look 😊

**Review by gupta-piyush19 — APPROVED** [review]

**Pull request merged**